apps/api/src/domain/models/digital-text/commands/add-photograph-to-digital-text-page/photograph-added-to-digital-text-page.event-handler.spec.ts

```diff
@@ -0,0 +1,193 @@
+import { ArangoDigitalTextQueryRepository } from '../../queries/arango-digital-text-query-repository';
+import {
+    DIGITAL_TEXT_QUERY_REPOSITORY_PROVIDER_TOKEN,
+    IDigitalTextQueryRepository,
+} from '../../queries/digital-text-query-repository.interface';
+import { ContentAddedToDigitalTextPage } from '../add-content-to-digital-text-page';
+import { PageAddedToDigitalText } from '../add-page-to-digital-text/page-added-to-digital-text.event';
+import { DigitalTextCreated } from '../digital-text-created.event';
+import { PhotographAddedToDigitalTextPage } from './photograph-added-to-digital-text-page.event';
+import { PhotographAddedToDigitalTextPageEventHandler } from './photograph-added-to-digital-text-page.event-handler';
+
+const digitalTextId = buildDummyUuid(12);
+
+const compositeId = {
+    type: AggregateType.digitalText,
+    id: digitalTextId,
+};
+
+const digitalTextTitle = 'The new book';
+
+const languageCodeForTitle = LanguageCode.English;
+
+const digitalTextCreated = new TestEventStream().andThen<DigitalTextCreated>({
+    type: 'DIGITAL_TEXT_CREATED',
+    payload: {
+        title: digitalTextTitle,
+        languageCodeForTitle,
+    },
+});
+
+const pageIdentifier = '20';
+
+const digitalTextPageAdded = digitalTextCreated
+    .andThen<PageAddedToDigitalText>({
+        type: 'PAGE_ADDED_TO_DIGITAL_TEXT',
+        payload: {
+            identifier: pageIdentifier,
+        },
+    })
+    .andThen<ContentAddedToDigitalTextPage>({
+        type: 'CONTENT_ADDED_TO_DIGITAL_TEXT_PAGE',
+        payload: {
+            pageIdentifier,
+            text: 'dummy page content',
+            languageCode: LanguageCode.English,
+        },
+    });
+
+const photographId = buildDummyUuid(2);
+
+const photographTitle = 'A photo of a dolphin';
+
+const photographCreated = new TestEventStream().andThen<PhotographCreated>({
+    type: 'PHOTOGRAPH_CREATED',
+    payload: {
+        title: photographTitle,
+        languageCodeForTitle,
+    },
+});
+
+const [photographCreationEvent] = photographCreated.as({
+    type: AggregateType.photograph,
+    id: photographId,
+}) as [PhotographCreated];
+
+const existingPhotographView = PhotographViewModel.fromPhotographCreated(photographCreationEvent);
+
+const photographAddedToDigitalTextPage =
+    digitalTextPageAdded.andThen<PhotographAddedToDigitalTextPage>({
+        type: 'PHOTOGRAPH_ADDED_TO_DIGITAL_TEXT_PAGE',
+        payload: {
+            aggregateCompositeIdentifier: { id: digitalTextId },
+            pageIdentifier,
+            photographId,
+        },
+    });
+
+const [creationEvent, pageAddedEvent, contentAddedToPageEvent, photographAddedToPageEvent] =
+    photographAddedToDigitalTextPage.as(compositeId) as [
+        DigitalTextCreated,
+        PageAddedToDigitalText,
+        ContentAddedToDigitalTextPage,
+        PhotographAddedToDigitalTextPage
+    ];
+
+const existingView = DigitalTextViewModel.fromDigitalTextCreated(creationEvent)
```

Comment:
I'd also use `buildTestInstance(DigitalTextViewModel,{...})` here. It's not clear that we will continue to use the `apply(...)` pattern. It's a vestige of a direction we decided not to go.